sgf/parser.go

```diff
@@ -0,0 +1,174 @@
+package sgf
+
+import (
+	"errors"
+	"io"
+)
+
+// A node is comprised of key-value pairs (might hold other info someday)
+type Node struct {
+	Fields map[string]string
```

Comment:
Typically, I've called these Properties.

Should be a list of string since SGF properties can be repeated.

Also, I would expect a Node to be a recursive type:

Children []*Node

Reply:
Sounds good, I've redone Node to have both children and parent (to traverse either direction, if needed)

Reply:
I also didn't realize the properties can be repeated.  Changed to `map[string][]string`